### PR TITLE
Add HTTP status to exception

### PR DESCRIPTION
**Pull request by bhelx**
Meant to address https://github.com/killbilling/recurly-java-library/issues/116

\cc @steelThread

You should be able to get the response code:

```java
apiException.getRecurlyError().getHttpStatusCode();
```

### CODE REVIEW

**Comment by jaaknt** [comment]
I fixed in my fork already and problem is that recurlyError can be null
                    RecurlyAPIError recurlyError = null;
                    try {
                        recurlyError = xmlMapper.readValue(payload, RecurlyAPIError.class);
                    } catch (Exception e) {
                        log.debug("Unable to extract error", e);
                        if (recurlyError == null) {
                           recurlyError = new RecurlyAPIError();
                        }
                    }
                    recurlyError.setStatusCode(response.getStatusCode());
                    throw new RecurlyAPIException(recurlyError);

**Comment by bhelx** [comment]
@jaaknt does that change work for you?

@pierre any opinions?

File: src/main/java/com/ning/billing/recurly/RecurlyClient.java [hunk 1]
```diff
@@ -1010,12 +1010,13 @@ private void doDELETE(final String resource) {
                     }
                     throw new TransactionErrorException(errors);
                 } else {
-                    RecurlyAPIError recurlyError = null;
+                    RecurlyAPIError recurlyError = new RecurlyAPIError();
```
Comment (bhelx) [inline src/main/java/com/ning/billing/recurly/RecurlyClient.java]: This has the consequence that the recurlyError is never null but I think it should be okay.

File: src/main/java/com/ning/billing/recurly/model/RecurlyAPIError.java [hunk 2]
```diff
@@ -57,12 +59,17 @@ public void setDetails(final String details) {
         this.details = details;
     }
 
+    public void setHttpStatusCode(final int httpStatusCode) { this.httpStatusCode = httpStatusCode; }
```
Comment (pierre) [inline src/main/java/com/ning/billing/recurly/model/RecurlyAPIError.java]: Nit: current code style has these expanded.

**Comment by pierre** [comment]
👍

**Comment by bhelx** [comment]
Thanks @pierre 

fixed formatting and rebased

**Comment by steelThread** [comment]
Any idea when this change will be pushed out to maven central?

**Comment by bhelx** [comment]
Thanks @pierre 👍

**Pull request merged**